Files/ViewModels/Bundles/BundleContainerViewModel.cs

```diff
@@ -0,0 +1,337 @@
+        /// </summary>
+        public ObservableCollection<BundleItemViewModel> Contents { get; private set; } = new ObservableCollection<BundleItemViewModel>();
+
+        private string bundleName = "DefaultBundle";
+        public string BundleName
+        {
+            get => bundleName;
+            set => SetProperty(ref bundleName, value);
+        }
+
+        private Visibility noBundleContentsTextVisibility;
+        public Visibility NoBundleContentsTextVisibility
+        {
+            get => noBundleContentsTextVisibility;
+            set => SetProperty(ref noBundleContentsTextVisibility, value);
+        }
+
+        private string bundleRenameText = string.Empty;
+        public string BundleRenameText
+        {
+            get => bundleRenameText;
+            set => SetProperty(ref bundleRenameText, value);
+        }
+
+        private Visibility bundleRenameVisibility = Visibility.Collapsed;
+        public Visibility BundleRenameVisibility
+        {
+            get => bundleRenameVisibility;
+            set => SetProperty(ref bundleRenameVisibility, value);
+        }
+
+        #endregion
+
+        #region Commands
+
+        public ICommand RemoveBundleCommand { get; set; }
+
+        public ICommand RenameBundleCommand { get; set; }
+
+        public ICommand RenameBundleConfirmCommand { get; set; }
+
+        public ICommand RenameTextKeyDownCommand { get; set; }
+
+        public ICommand DragOverCommand { get; set; }
+
+        public ICommand DropCommand { get; set; }
+
+        #endregion
+
+        #region Constructor
+
+        public BundleContainerViewModel(IShellPage associatedInstance)
+        {
+            this.associatedInstance = associatedInstance;
+            this.BundleRenameText = BundleName;
+
+            // Create commands
+            RemoveBundleCommand = new RelayCommand(RemoveBundle);
+            RenameBundleCommand = new RelayCommand(RenameBundle);
+            RenameBundleConfirmCommand = new RelayCommand(RenameBundleConfirm);
+            RenameTextKeyDownCommand = new RelayCommand<KeyRoutedEventArgs>(RenameTextKeyDown);
+            DragOverCommand = new RelayCommand<DragEventArgs>(DragOver);
+            DropCommand = new RelayCommand<DragEventArgs>(Drop);
+        }
+
+        #endregion
+
+        #region Command Implementation
+
+        private void RemoveBundle()
+        {
+            if (WidgetsSettings.SavedBundles.ContainsKey(BundleName))
+            {
+                Dictionary<string, List<string>> allBundles = WidgetsSettings.SavedBundles; // We need to do it this way for Set() to be called
+                allBundles.Remove(BundleName);
+                WidgetsSettings.SavedBundles = allBundles;
+                NotifyItemRemoved(this);
+            }
+        }
+
+        private void RenameBundle()
+        {
+            if (BundleRenameVisibility == Visibility.Visible)
+                BundleRenameVisibility = Visibility.Collapsed;
```

Comment:
Add brackets, please.

Reply:
Sorry that I've missed some. I'm not used to this and my eyes just ignore them 😅